src/bepasty/app.py

```diff
@@ -45,13 +46,29 @@ def __call__(self, environ, start_response):
             return ['This URL does not belong to the bepasty app.'.encode()]
 
 
+def setup_secret_key(app):
+    """
+    to detect the configuration change, and make obsolete session for
+    old config, this add unique id to SECRET_KEY additionally.
+
+    NOTE: for now, only permissions is checked
+    """
+    # if app.config['SECRET_KEY'] is empty, keep as NullSession
+    if app.config['SECRET_KEY']:
+        perms = sorted(k + v for k, v in app.config['PERMISSIONS'].items())
+        perms = ''.join(perms).encode()
+        app.config['SECRET_KEY'] += hashlib.sha256(perms).hexdigest()
+
+
 def create_app():
     app = Flask(__name__)
 
     app.config.from_object('bepasty.config.Config')
     if os.environ.get('BEPASTY_CONFIG'):
         app.config.from_envvar('BEPASTY_CONFIG')
 
+    setup_secret_key(app)
```

Comment:
hopefully this is early enough, before flask actually does anything with the secret.

Reply:
ok